modules/tinymce/src/core/main/ts/delete/MergeBlocks.ts

```diff
@@ -78,9 +78,9 @@ const findInsertionPoint = (toBlock: SugarElement<Element>, block: SugarElement<
 const getInsertionPoint = (fromBlock: SugarElement<Element>, toBlock: SugarElement<Element>): Optional<SugarElement> =>
   Compare.contains(toBlock, fromBlock) ? findInsertionPoint(toBlock, fromBlock) : Optional.none();
 
-const trimBr = (first: boolean, block: SugarElement) => {
+const trimBr = (first: boolean, block: SugarElement<Element>) => {
   CaretFinder.positionIn(first, block.dom)
-    .map((position) => position.getNode())
+    .bind((position) => Optional.from(position.getNode()))
```

Comment:
Here's an example of one of the cases I did change to make sure to account for the fact that `CaretPosition.getNode()` can returned `undefined` in specific cases.